File: swift/입문/중복된_문자_제거.swift
// 중복된 문자 제거 문제

import Foundation

func solution(_ my_string:String) -> String {
    var result = ""
    for s in my_string {
        if !result.contains(String(s)) {
            result += String(s)
        }
    }
    return result
}

// Set.joined() : Set은 Character들의 집합이므로 직접 joined()를 사용할 수 없습니다.

File: python/입문/중복된_문자_제거.py
# 중복된 문자 제거 문제

def solution(my_string):
    answer = ''
    for s in my_string:
        if not s in answer:
            answer += s
    return answer

# dict.fromkeys(my_string)
# python 3.7 이상에서 딕셔너리가 삽입 순서를 기억
# 딕셔너리 특징
# 딕셔너리를 join() 하면 딕셔너리 키들만 연결